src/averageAges.js

```diff
@@ -36,8 +50,25 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredTab = withChildren
+    ? people.filter(isMother)
+    : people.filter(isWoman);
+
+  function isWoman(person) {
+    return person.sex === 'f';
+  };
+
+  function isMother(person) {
+    return people.some(el => el.mother === person.name);
+  };
+
+  function sumAllAges(prev, next) {
+    return prev + (next.died - next.born);
+  };
```

Comment:
You could create helper function for calculating average as you are using this in multiple places